lib/fluent/plugin/exception_detector.rb

```diff
@@ -223,14 +223,22 @@ def push(time_sec, record)
       force_flush if @max_lines > 0 && @messages.length == @max_lines
     end
 
+    def combined_message
```

Comment:
I'm not happy with defining a method for this -- too much Ruby magic. You are essentially injecting newlines after trace lines that don't have them, right? Why not do it as a comprehension, e.g.:
```
combined_message = @messages.inject([]) {|r, e| r << e; r << $RS unless e.empty? || e.end_with?("\n"); r}.join
```